### ISSUE TITLE
[BUG]: Rocks sync keeps installing and uninstalling plugins if the name is not lower case

### ISSUE THREAD

**Issue by rzlink** [nvim-neorocks/rocks.nvim#287]
Here is the rocks log:
``` 
INFO | 2024-04-23 20:26:27 | ...share/nvim/rocks/rocks_rtp/lua/rocks/operations/init.lua:88 | syncing...
INFO | 2024-04-23 20:26:27 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "list", "--porcelain" }
INFO | 2024-04-23 20:26:27 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:39 | Installing: comment.nvim -> 0.8.0
INFO | 2024-04-23 20:26:27 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "--server='https://luarocks.org/manifests/neorocks/'", "--server='https://nvim-neorocks.github.io/rocks-binaries/'", "install", "comment.nvim", "0.8.0" }
INFO | 2024-04-23 20:26:28 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:84 | Installed: comment.nvim -> 0.8.0
INFO | 2024-04-23 20:26:28 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:39 | Installing: luasnip -> 2.3.0
INFO | 2024-04-23 20:26:28 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "--server='https://luarocks.org/manifests/neorocks/'", "--server='https://nvim-neorocks.github.io/rocks-binaries/'", "install", "luasnip", "2.3.0" }
INFO | 2024-04-23 20:26:29 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:84 | Installed: luasnip -> 2.3.0
INFO | 2024-04-23 20:26:29 | .../.local/share/nvim/rocks/rocks_rtp/lua/rocks-git/git.lua:34 | { "git", "fetch" }
INFO | 2024-04-23 20:26:29 | .../.local/share/nvim/rocks/rocks_rtp/lua/rocks-git/git.lua:34 | { "git", "checkout", "compat-nvim-0.6", "--force", "--recurse-submodules" }
INFO | 2024-04-23 20:26:30 | .../.local/share/nvim/rocks/rocks_rtp/lua/rocks-git/git.lua:34 | { "git", "fetch" }
INFO | 2024-04-23 20:26:30 | .../.local/share/nvim/rocks/rocks_rtp/lua/rocks-git/git.lua:34 | { "git", "checkout", "v0.8.0", "--force", "--recurse-submodules" }
....
INFO | 2024-04-23 20:26:34 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "show", "--porcelain", "comment.nvim" }
INFO | 2024-04-23 20:26:34 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:109 | Uninstalling: comment.nvim
INFO | 2024-04-23 20:26:34 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "remove", "comment.nvim" }
INFO | 2024-04-23 20:26:34 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:126 | Uninstalled: comment.nvim
INFO | 2024-04-23 20:26:34 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "list", "--porcelain" }
....
INFO | 2024-04-23 20:26:38 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "show", "--porcelain", "luasnip" }
INFO | 2024-04-23 20:26:38 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:109 | Uninstalling: luasnip
INFO | 2024-04-23 20:26:38 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "remove", "luasnip" }
INFO | 2024-04-23 20:26:38 | ...re/nvim/rocks/rocks_rtp/lua/rocks/operations/helpers.lua:126 | Uninstalled: luasnip
INFO | 2024-04-23 20:26:38 | ....local/share/nvim/rocks/rocks_rtp/lua/rocks/luarocks.lua:85 | { "luarocks", "--lua-version=5.1", "--tree=/home/rzlink/.local/share/nvim/rocks", "list", "--porcelain" }



```

**Comment by mrcjkb** [comment]
Hey :wave: 

Thanks for reporting.
Could you please post your rocks.toml?

**Comment by rzlink** [comment]
it's on windows WSL. 

`
[rocks]
[plugins]
"rocks.nvim" = "2.24.0"
"rocks-config.nvim" = "1.5.0"
"rocks-git.nvim" = "1.4.0"
"rocks-treesitter.nvim" = "scm"
neorg = "8.4.0"
"flash.nvim" = "1.18.2"
"telescope.nvim" = "scm"
"which-key.nvim" = "1.6.0"
"tokyonight.nvim" = "3.0.1"
"conform.nvim" = "5.6.0"
"todo-comments.nvim" = "1.2.0"
"indent-blankline.nvim" = "3.5.4"
"noice.nvim" = "2.0.1"
nvim-navic = "scm"
"mini.diff" = "scm"
nvim-web-devicons = "0.99"
nvim-lspconfig = "0.1.7"
"lazygit.nvim" = "scm"
"toggleterm.nvim" = "2.11.0"
"copilot.lua" = "scm"
nvim-treesitter = "0.9.2"
"mason.nvim" = "1.10.0"
"mason-lspconfig.nvim" = "1.27.0"
"mason-tool-installer.nvim" = "scm"
"Comment.nvim" = "0.8.0"
nvim-autopairs = "scm"
cmp-nvim-lsp = "scm"
cmp-buffer = "scm"
cmp-path = "scm"
cmp-cmdline = "scm"
LuaSnip = "2.3.0"
nvim-cmp = "0.0.1"
tree-sitter-toml = "scm"
"mini.nvim" = "0.9.0"
"mini.files" = "0.12.0"
tree-sitter-rust = "scm"
"trouble.nvim" = "2.10.0"
"gruvbox.nvim" = "2.0.0"
toml-edit = "0.3.6"
github-nvim-theme = "1.0.1"

[plugins."gitsigns.nvim"]
git = "lewis6991/gitsigns.nvim"
rev = "v0.8.0"

[plugins.nvim-navbuddy]
git = "SmiteshP/nvim-navbuddy"

[plugins.vim-translator]
git = "voldikss/vim-translator"

[plugins."lualine.nvim"]
git = "nvim-lualine/lualine.nvim"
rev = "compat-nvim-0.6"

[config]
plugins_dir = "plugins/"
auto_setup = false

[treesitter]
auto_highlight = [
  "haskell",
  "dhall",
  "rust",
  "toml",
  "lua"
]
auto_install = "prompt" # true | false
`

`
==============================================================================
rocks: require("rocks.health").check()

Checking external dependencies ~
- OK luarocks: found /usr/bin/luarocks 3.8.0
- OK lua: found Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio

Checking config ~
- OK No errors found in config.


`

**Comment by mrcjkb** [comment]
Thanks.

And it's only luasnip and comment.nvim that get uninstalled?

**Comment by mrcjkb** [comment]
Ah, I think I see what the issue is.
luarocks packages are always lower case. Because those two entries have upper case names, rocks.nvim doesn't recognise them as installed.

I'll see if I can implement a fix.

**Comment by mrcjkb** [comment]
@rzlink I've implemented a fix (#288 - see the PR description).

To fix this on your end, you will need to change the rocks.toml entries to be lowercase.

**Comment by rzlink** [comment]
Thanks @mrcjkb for quick fix !!!
